libnavigation-core/src/main/java/com/mapbox/navigation/core/navigator/LocationEx.kt

```diff
@@ -48,11 +51,48 @@ internal fun Location.toFixLocation(): FixLocation {
         provider,
         bearingAccuracy,
         speedAccuracy,
-        verticalAccuracy
+        verticalAccuracy,
+        if (extras != null) extras.toMap() else Bundle().toMap()
     )
 }
 
 internal fun List<FixLocation>.toLocations(): List<Location> = this.map { it.toLocation() }
 
 private fun isCurrentSdkVersionEqualOrGreaterThan(sdkCode: Int): Boolean =
     Build.VERSION.SDK_INT >= sdkCode
+
+internal fun HashMap<String, Value>.toBundle(): Bundle? {
+    if (this.isEmpty()) return null
+    val bundle = Bundle()
+    for ((key, value) in this.entries) {
+        when (val contents = value.contents) {
+            is Double -> bundle.putDouble(key, contents)
+            is Long -> bundle.putLong(key, contents)
+            is Boolean -> bundle.putBoolean(key, contents)
+            is String -> bundle.putString(key, contents)
+            else -> {
+                // do nothing
+            }
```

Comment:
It'd be great if these https://github.com/mapbox/mapbox-maps-android/blob/main/extension-style/src/main/java/com/mapbox/maps/extension/style/utils/TypeUtils.kt were exposed internally so we don't have to implement them on each SDK.

cc @mr1sunshine